src/pages/actions/[slug].tsx

```diff
@@ -280,7 +282,7 @@ export default function ActionPage() {
           />
         </Container>
       )}
-      {causalNodes.length > 0 && (
+      {causalNodes.length > 0 && !theme?.settings.hideActionGrid && (
```

Comment:
Also when using `useTheme` the value will always be defined, so the `?` won't be needed.